### PR TITLE
[NUI] Add FindLayerByID(), FindDescendantByID()

### PR DESCRIPTION
**Pull request by dongsug-song**
### Description of Change ###
[NUI] Add FindLayerByID(), FindChildByID()
- FindLayerByID() in Window and FindChildByID() in View are added.
- These are all Hidden-APIs.

### API Changes ###
none

### CODE REVIEW

**Comment by huiyueun** [comment]
Do you need to add it to API9?

File: src/Tizen.NUI/src/public/BaseComponents/ViewPublicMethods.cs [hunk 1]
```diff
@@ -632,5 +632,22 @@ public void ObjectDump()
...
         }
+
+        /// <summary>
+        /// Search through this View's hierarchy for a View with the given unique ID.
+        /// </summary>
+        /// <param name="id">The ID of the View to find.</param>
+        /// <remarks>Hidden-API</remarks>
+        /// <returns>A handle to the View if found, or an empty handle if not.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public View FindChildByID(uint id)
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/ViewPublicMethods.cs:643]: I wonder if child includes grandchildren, if true how about use `Descendant` instead child
```suggestion
        public View FindDescendantByID(uint id)
```

File: src/Tizen.NUI/src/public/Layer.cs [hunk 2]
```diff
@@ -672,6 +687,14 @@ internal void SetWindow(Window win)
             window = win;
         }
 
+        internal uint GetId()
```
Comment (myroot) [inline src/Tizen.NUI/src/public/Layer.cs:690]: Why not private?
```suggestion
        private uint GetId()
```

**Comment by dongsug-song** [comment]
> Do you need to add it to API9?

yes, after this is merged in API8, then I will upload a PR on API9 too.
this is urgent request from VD and it's in tizen_6.0

File: src/Tizen.NUI/src/public/BaseComponents/ViewPublicMethods.cs [hunk 3]
```diff
@@ -632,5 +632,22 @@ public void ObjectDump()
...
+        /// <remarks>Hidden-API</remarks>
+        /// <returns>A handle to the View if found, or an empty handle if not.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public View FindDescendantByID(uint id)
+        {
+            //to fix memory leak issue, match the handle count with native side.
+            IntPtr cPtr = Interop.Actor.Actor_FindChildById(swigCPtr, id);
+            View ret = this.GetInstanceSafely<View>(cPtr);
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
```
Comment (myroot) [inline src/Tizen.NUI/src/public/BaseComponents/ViewPublicMethods.cs:649]: You do better to use existed method, or I recommend to remove `internal FindChildById` and replace to this API
```suggestion
            return FindChildById(id);
```

Comment (myroot) [reply src/Tizen.NUI/src/public/BaseComponents/ViewPublicMethods.cs:649]: Duplicated code and APIs

https://github.com/Samsung/TizenFX/blob/7a14e7532a6622108bc0119764a964475f371773/src/Tizen.NUI/src/public/BaseComponents/ViewInternal.cs#L499-L506

https://github.com/Samsung/TizenFX/blob/7a14e7532a6622108bc0119764a964475f371773/src/Tizen.NUI/src/public/BaseComponents/ViewInternal.cs#L508-L511

**Review by Seoyeon2Kim — APPROVED** [review]
Looks good to me!
Especially, adding a sample related to this PR is a great idea.

**Pull request merged**